troubleshooting.rst

```diff
@@ -0,0 +1,403 @@
+
+
+If you prefer to consume metrics via logs, the ``telemetry.log``
+(:zeek:see:`Telemetry::Info`) may work. Its
+format is a bit unusual, however. See the :ref:`framework-telemetry`'s
+documentation for more details about the log and how to add further metrics
+from your own Zeek scripts.
+
+.. _Prometheus server: https://prometheus.io/
+
+
+stats.log
+---------
+
+The ``stats.log`` is enabled when loading the :doc:`/scripts/policy/misc/stats.zeek` script.
+This is the default with the stock ``local.zeek`` included with Zeek. This
+log provides stats about Zeek's operational behavior in a structured log format.
+
+See the :zeek:see:`Stats::Info` record documentation for a description of
+the individual fields.
+
+The default reporting interval is 5 minutes. It can make sense to reduce
+this interval for testing or during troubleshooting via
+``redef Stats::report_interval=30sec``. Stats collection may have a
+non-negligible impact on performance and running, for example,
+every second may be detrimental to performance.
+
+For historic reasons, this log contains delta values for ``pkts_proc``,
+``bytes_recv``, ``events_proc``, ``tcp_conns``, etc. This can make it
+difficult to use the values as is in metrics systems that expect counter
+metrics to continuously grow and compute rates or delta values on the fly.
+
+.. note::
+
+   If you're creating your own custom metrics or stats-like log, consider
+   using absolute values for counter metrics. Relative values can
+   always be derived from two absolute values. The inverse is not true.
+   Popular metrics systems usually assume absolute counter values, too.
+
+Following an example of a ``stats.log`` entry:
+
+.. code-block:: console
+
+   $ zeek -C -i eth0 local Stats::report_interval=30sec LogAscii::use_json=T
+   $ jq < stats.log
+   ...
+   {
+       "ts": 1684828680.616951,
+       "peer": "zeek",
+       "mem": 344,
+       "pkts_proc": 300000,
+       "bytes_recv": 78092228,
+       "pkts_dropped": 0,
+       "pkts_link": 299609,
+       "pkt_lag": 0.003422975540161133,
+       "events_proc": 448422,
+       "events_queued": 448422,
+       "active_tcp_conns": 2279,
+       "active_udp_conns": 2809,
+       "active_icmp_conns": 96,
+       "tcp_conns": 6747,
+       "udp_conns": 5954,
+       "icmp_conns": 48,
+       "timers": 67510,
+       "active_timers": 35086,
+       "files": 8165,
+       "active_files": 0,
+       "dns_requests": 218,
+       "active_dns_requests": 2,
+       "reassem_tcp_size": 7816,
+       "reassem_file_size": 0,
+       "reassem_frag_size": 0,
+       "reassem_unknown_size": 0
+   }
+
+prof.log
+--------
+
+The ``prof.log`` provides aggregated information about Zeek's runtime status
+in a fairly non-structured text format.
+Likely future metrics will be added through the Telemetry framework mentioned
+above, but as of now it does contain information about queue sizes within
+the threading subsystem and other details that are not yet exposed otherwise.
```

Comment:
This sounds like something that a ticket could be opened for as a future enhancement.